guides/testing-micronaut-kafka-listener-using-testcontainers/testing-micronaut-kafka-listener-using-testcontainers.adoc

```diff
@@ -0,0 +1,188 @@
+common:header-top.adoc[]
+
+In this guide you will learn how to
+
+* Create a Micronaut application with https://www.confluent.io/lp/apache-kafka/[Kafka] integration.
+* Implement a Kafka Listener and persist data in https://www.mysql.com[MySQL] database
+* Test the Kafka Listener using https://testcontainers.com[Testcontainers] and http://www.awaitility.org[Awaitility].
+
+common:requirements.adoc[]
+
+common:docker-testcontainers.adoc[]
+
+common:completesolution.adoc[]
+
+== What we are going to achieve in this guide
+We are going to create a Micronaut project with *Kafka*, *Micronaut Data JPA* and *MySQL*,
+where we implement a Kafka Listener which receives an event payload and persists the event data in the database.
+Then we will test this Kafka Listener using the Testcontainers Kafka
+and MySQL modules in conjunction with http://www.awaitility.org/[Awaitility].
+
+
+The generated application has http://www.awaitility.org[Awaitility] library as test dependency
+which we can use for asserting the expectations of an asynchronous process flow.
+
+common:create-app-features-testcontainers.adoc[]
+
+== Testcontainers Dependencies
+
+The generated application contains the following Testcontainers dependencies:
+
+dependency:testcontainers[groupId=org.testcontainers,scope=test]
```

Comment:
You don't need to add "org.testcontainers:testcontainers" dependency explicitly. Adding the specific modules like kafka, MySQL etc is sufficient.

Reply:
resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1332/commits/922664aa013ceedd232d57b45935f36b74d410b5